### ISSUE TITLE
[EmailSearching] [Advance Search] Advance Search with Has attachment condition is not working correctly

### ISSUE THREAD

**Issue by QuangHoang1210** [linagora/tmail-flutter#1536]
OS: Andoird / Ios / Window
Device: Mobile / Laptop
TMail version: 0.7.1 - Production

*Precondition: User already login successfully

*Reproduce Step:
1, User tap on searchbox and input "dat" on textbox to search --> Get result all email contain "dat" and there are 2 email have attachment file.
2, User add one more searching condition by clicking Has attachment checkbox 

Actual: System display no email on searching result
Expected: It should display 2 email which have attachment file in this case <following video bellow> 

https://user-images.githubusercontent.com/124866146/223047044-aa87e2d4-f94c-410b-8ac0-10d9f0cc9050.mp4

**Comment by chibenwa** [comment]
Please show the underlying JMAP query + the actual response.

**Comment by QuangHoang1210** [comment]
I'm trying again. Idk why now it's simple ok. !_!
Try to reproduce

**Comment by QuangHoang1210** [comment]
1, User input "dat" in search text box 
Request payload: 
```
{
  "using": [
    "urn:ietf:params:jmap:core",
    "urn:ietf:params:jmap:mail"
  ],
  "methodCalls": [
    [
      "Email/query",
      {
        "accountId": "912618096e03fb03fe14293a97102272d4875bd9c3d80ad49de32c58bb4b28bd",
        "filter": {
          "operator": "AND",
          "conditions": [
            {
              "before": "2023-02-09T08:44:45.000Z",
              "text": "dat"
            }
          ]
        },
        "sort": [
          {
            "isAscending": false,
            "property": "receivedAt"
          }
        ],
        "limit": 20
      },
      "c0"
    ],
    [
      "Email/get",
      {
        "accountId": "912618096e03fb03fe14293a97102272d4875bd9c3d80ad49de32c58bb4b28bd",
        "#ids": {
          "resultOf": "c0",
          "name": "Email/query",
          "path": "/ids/*"
        },
        "properties": [
          "id",
          "subject",
          "from",
          "to",
          "cc",
          "bcc",
          "keywords",
          "size",
          "receivedAt",
          "sentAt",
          "preview",
          "hasAttachment",
          "replyTo",
          "mailboxIds"
        ]
      },
      "c1"
    ]
  ]
}
```

**Comment by dab246** [comment]
Fixed at PR #1524
